packages/medusa/src/api/admin/claims/middlewares.ts

```diff
@@ -0,0 +1,229 @@
+      validateAndTransformQuery(
+        AdminGetOrdersOrderParams,
+        QueryConfig.retrieveTransformQueryConfig
+      ),
+    ],
+  },
+  {
+    method: ["POST"],
+    matcher: "/admin/claims/:id/inbound/shipping-method/:action_id",
+    middlewares: [
+      validateAndTransformBody(AdminPostClaimsShippingActionReqSchema),
+      validateAndTransformQuery(
+        AdminGetOrdersOrderParams,
+        QueryConfig.retrieveTransformQueryConfig
+      ),
+    ],
+  },
+  {
+    method: ["DELETE"],
+    matcher: "/admin/claims/:id/inbound/shipping-method/:action_id",
+    middlewares: [
+      validateAndTransformQuery(
+        AdminGetOrdersOrderParams,
+        QueryConfig.retrieveTransformQueryConfig
+      ),
+    ],
+  },
+
+  {
+    method: ["POST"],
+    matcher: "/admin/claims/:id/outbound/items",
+    middlewares: [
+      validateAndTransformBody(AdminPostClaimsAddItemsReqSchema),
+      validateAndTransformQuery(
+        AdminGetOrdersOrderParams,
+        QueryConfig.retrieveTransformQueryConfig
+      ),
+    ],
+  },
+  {
+    method: ["POST"],
+    matcher: "/admin/claims/:id/outbound/items/:action_id",
+    middlewares: [
+      validateAndTransformBody(AdminPostClaimsRequestItemsActionReqSchema),
+      validateAndTransformQuery(
+        AdminGetOrdersOrderParams,
+        QueryConfig.retrieveTransformQueryConfig
+      ),
+    ],
+  },
+  {
+    method: ["DELETE"],
+    matcher: "/admin/claims/:id/outbound/items/:action_id",
+    middlewares: [
+      validateAndTransformQuery(
+        AdminGetOrdersOrderParams,
+        QueryConfig.retrieveTransformQueryConfig
+      ),
+    ],
+  },
+  {
+    method: ["POST"],
+    matcher: "/admin/claims/:id/outbound/shipping-method",
+    middlewares: [
+      validateAndTransformBody(AdminPostClaimsShippingReqSchema),
+      validateAndTransformQuery(
+        AdminGetOrdersOrderParams,
+        QueryConfig.retrieveTransformQueryConfig
+      ),
+    ],
+  },
+  {
+    method: ["POST"],
+    matcher: "/admin/claims/:id/outbound/shipping-method/:action_id",
+    middlewares: [
+      validateAndTransformBody(AdminPostClaimsShippingActionReqSchema),
+      validateAndTransformQuery(
+        AdminGetOrdersOrderParams,
+        QueryConfig.retrieveTransformQueryConfig
+      ),
+    ],
+  },
+  {
+    method: ["DELETE"],
+    matcher: "/admin/claims/:id/outbound/shipping-method/:action_id",
+    middlewares: [
+      validateAndTransformQuery(
+        AdminGetOrdersOrderParams,
+        QueryConfig.retrieveTransformQueryConfig
+      ),
+    ],
+  },
+
+  {
+    method: ["POST"],
+    matcher: "/admin/claims/:id/request",
+    middlewares: [
+      validateAndTransformBody(AdminPostClaimsConfirmRequestReqSchema),
+      validateAndTransformQuery(
+        AdminGetOrdersOrderParams,
+        QueryConfig.retrieveTransformQueryConfig
+      ),
+    ],
+  },
+  {
+    method: ["DELETE"],
+    matcher: "/admin/claims/:id/request",
```

Comment:
nit: Typo + I don't think the route implementation is included
```suggestion
    matcher: "/admin/claims/:id",
```

Reply:
Also `POST /admin/claims/:id/request` isn't included. (Disregard if it will be added in part 2 :D).

Reply:
Claims will have a DELETE method to cancel the "draft claim" but it also needs a DELETE to cancel the claim after it is created and confirmed.